### PR TITLE
Add example producer using DeferredConfirm

### PR DESCRIPTION
**Pull request by Zerpet**
The usage of `DeferredConfirmation` may not be trivial or clear to everybody.
This example demonstrates the use case to send a message and wait for its
confirmation, before moving forward. This would probably provide suboptimal
performance, but not every user cares about having a blazing fast rabbit.

### CODE REVIEW

**Review by lukebakken — CHANGES_REQUESTED** [review]
It's good to have a deferred confirmation example, but if we show how to do it this way, people will copy this code and wonder why publishing is slow.

We should re-work this to do it the "right" way in the same manner that we do for our ["Tutorial 7" examples](https://www.rabbitmq.com/tutorials/tutorial-seven-java.html), then we can add Go to that.

**Comment by lukebakken** [comment]
@Zerpet let me know what you think

File: _examples/producer-confirms/producer.go [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+					exitConfirmHandler(confirms, done_ch)
+					return
+				default:
+					select {
+					case publish_ch <- struct{}{}:
+						Log.Println("confirm handler: sent OK to publish")
+					case <-time.After(time.Second * 5):
+						WarnLog.Println("confirm handler: timeout indicating OK to publish (this should never happen!)")
+					}
+				}
```
Comment (Zerpet) [inline _examples/producer-confirms/producer.go:201]: Do we need the `default` case to then `select`?

I think we can just add the cases `case publish_ch <- struct{}{}` and `case <-time.After(time.Second * 5)` in the outter `select`, and it should behave the same?

**Review by Zerpet — COMMENTED** [review]
One question regarding the `select`. Everything else looks good. I pushed a change to follow Go naming [convention](https://go.dev/doc/effective_go#mixed-caps) 👀 

What do you think @lukebakken ?

File: _examples/producer-confirms/producer.go (hunk 1 above)
Comment (lukebakken) [reply _examples/producer-confirms/producer.go:201]: Ah yep, that was left over from when `publishCh` was unbuffered.

**Pull request merged**